### PR TITLE
Cbrune/bfd 681 - Tax Num to v1 and v2

### PR DESCRIPTION
**Pull request by cbrunefearless**
### Change Details
Added tax number to v1 and v2.  Changes the apply method to a transform method, so the change could persist down to the transformer.

### Acceptance Validation

Make sure the tax number is correctly mapped in v1 and v2.  I was told v2 was an extension.

### Feedback Requested
Look at the mapping of the tax_num.. Run the application in v2 with a header of iIncludeTaxNumbers" of true and should return the tax number and a header of IncludeTaxNumbers" of false

### External References

- [BFD-681](https://jira.cms.gov/browse/BFD-681)

### Security Implications

- [ ] new software dependencies

N/A
- [ ] altered security controls

N/A
- [ ] new data stored or transmitted

N/A
- [ ] security checklist is completed for this change

N/A
- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/CarrierClaimTransformer.java
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/CarrierClaimTransformer.java:171]: Not sure this FIXME belongs.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java
Comment (Dranith) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java:181]: This can be simplified to just:
```
return v.contains("true");
```

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java
Comment (Dranith) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java:247]: I'm double checking on this.  The mapping documents for Carrier and DME still say "Not Mapping this field" so I'd like to get Jack to update before we move forward with this.

**Review by mackec1 — DISMISSED** [review]
LGTM

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java:181]: fixed

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java:247]: i agree

Comment (Dranith) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java:247]: Mapping sheet has been updated.  This is all good.  Resolving.

**Review by Dranith — APPROVED** [review]
Looks good.

**Review by jarwilliams1971 — DISMISSED** [review]
looks good

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/RequestHeaders.java:224]: Couldn't we simplify this as:

return (headerValue != null && headerValue.equalsIgnoreCase("TRUE"));

**Review by mackec1 — APPROVED** [review]
LGTM; the comment I entered is a code problem but a comment that it could be simplified.

**Review by mackec1 — APPROVED** [review]
ship it!

**Review by mackec1 — APPROVED** [review]

**Review by Dranith — APPROVED** [review]
re-approving after merge

**Pull request merged**
